packages/components/src/useScreenSize/useScreenSize.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useRef, useState } from 'react';
+
+import getVariant from './getVariant';
+
+const useScreen = () => {
+    const [ variant, setVariant ] = useState(() => getVariant());
+    const prev = useRef(variant);
+
+    useEffect(() => {
+        function handleResize() {
+            const newVariant = getVariant();
+            if (newVariant !== prev.current) {
+                prev.current = newVariant;
```

Comment:
You have to store function reference here, instead of the actual value, otherwise next cycle will not calculate with new screen size.
```suggestion
            const newVariant = getVariant;
            if (newVariant() !== prev.current()) {
                prev.current = newVariant;
```

Reply:
Why to store the function reference?

It's just a string and it compares two strings.

If you store a function in the ref, you would get always true because both of the functions are the same.

Reply:
Then the state initialization is incorrect, as you are storing the callback to it.

Reply:
Nope, passing a function to useState hooks is a init function - a function, that runs only once on the initial render and returns the initial state.

If you want to store a function in state, you have to do `useState(() => () => blabla())`

Reply:
@karelhala it's not. You can use a function to initialize `useState`. It will use the return value.
https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
Good to know!